### DISCUSSION TITLE
Best way to work with JS events?

### DISCUSSION THREAD

**Discussion by niamleeson** [LegendApp/legend-state#307] [Q&A]
Hello,

After reading the docs, I'm not completely clear on how legend-state can be used with fine controlling state updates from keyboard/mouse events. Would you kindly elaborate more on this topic?

Thank you!

**Comment by jmeistrich** [comment]
I'm not quite sure what you mean. You can set observables from events to trigger re-renders. Are you trying something that's not working?
